zbMATH.js

```diff
@@ -35,105 +35,163 @@
 */
 
 
-function detectWeb(doc, _url) {
-	if (ZU.xpath(doc, '//div[@class="list"]/article').length > 0) {
+function detectWeb(doc) {
+	if (doc.querySelector(".content-result")) { // search results present
 		return "multiple";
 	}
-	else if (ZU.xpath(doc, '//a[contains(@class, "bib")]').length > 0) { // contains
+	else if (doc.querySelector(".bib")) {
 		// it is a single entry --> generic fallback = journalArticle
 		return "journalArticle";
 	}
 	return false;
 }
 
-function scrape(doc, _url) {
-	var bibArray = doc.getElementsByClassName("bib");
-	var bibUrl = bibArray[0].getAttribute('href');// e.g. "bibtex/06115874.bib"
+function getSearchResults(doc, checkOnly) {
+	let items = {};
+	let found = false;
+	let rows = doc.querySelectorAll(".content-result .list > article");
+	for (let row of rows) {
+		let href = attr(row, ".title a", "href");
+		let title = ZU.trimInternal(text(row, ".title"));
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found && items;
+}
 
-	ZU.doGet(bibUrl, function (text) {
-		// Z.debug(text);
-		
-		var trans = Zotero.loadTranslator('import');
-		trans.setTranslator('9cb70025-a888-4a29-a210-93ec52da40d4');// https://github.com/zotero/translators/blob/master/BibTeX.js
-		trans.setString(text);
+async function doWeb(doc, url) {
+	if (detectWeb(doc, url) == 'multiple') {
+		let items = await Zotero.selectItems(getSearchResults(doc, false));
+		if (!items) return;
+		for (let url of Object.keys(items)) {
+			await scrape(await requestDocument(url));
+		}
+	}
+	else {
+		await scrape(doc, url);
+	}
+}
 
-		trans.setHandler('itemDone', function (obj, item) {
-			item.title = item.title.replace(/\.$/, '');
-			
-			if (item.publisher) {
-				var publisherSeparation = item.publisher.indexOf(":");
-				if (publisherSeparation != -1) {
-					item.place = item.publisher.substr(0, publisherSeparation);
-					item.publisher = item.publisher.substr(publisherSeparation + 1);
-				}
+async function scrape(doc, url) {
+	let bibURL = attr(doc, ".bib", "href");
+	if (!bibURL) {
+		Z.debug(`Error: document at ${url} does not contain bibTeX link`);
+		return;
+	}
+	let bibTeXDoc = await ZU.requestText(bibURL);
```

Comment:
```suggestion
	let bibTeXDoc = await requestText(bibURL);
```
No need for `ZU.` for the newer translator utility functions